src/ngraph_emitter.cc

```diff
@@ -1604,6 +1608,92 @@ void Emitter::CreateLayerOps() {
     }
     return conv;
   };
+
+  ngraph_op_funcs_["UpSampling"] =
+      [this](const NodePtr& node) -> NgraphNodePtr {
+
+    int scale = get_default(node, "scale", 1);
+    std::string sample_type =
+        get_default(node, "sample_type", std::string("nearest"));
+    std::string multi_input_mode =
+        get_default(node, "multi_input_mode", std::string("concat"));
+
+    NgraphNodePtr result{nullptr};
+    NgraphNodePtr input = op_map_.at(node->inputs_[0]);
+    if (sample_type == "nearest") {
+      auto upsample_nearest = [](const NgraphNodePtr& data,
+                                 const int scale) -> NgraphNodePtr {
+        const auto& in_shape = data->get_shape();
+        auto out_shape = in_shape;
+        out_shape[2] *= scale;
+        out_shape[3] *= scale;
+        auto create_nn_matrix = [scale](const size_t row_size,
+                                        const size_t col_size) {
+          std::vector<float> nn_matrix(row_size * col_size, 0);
+          for (size_t i = 0; i < row_size; ++i) {
+            for (size_t j = i * scale; j < (i + 1) * scale; ++j) {
+              nn_matrix[i * col_size + j] = 1;
+            }
+          }
+          return nn_matrix;
+        };
+        auto nn_matrix_row = create_nn_matrix(in_shape[3], out_shape[3]);
+        auto nn_matrix_col = create_nn_matrix(in_shape[2], out_shape[2]);
```

Comment:
Does the MXNet op only work with 2D images?

Reply:
Yes, mxnet expects 4D input.